numba/cuda/tests/cudapy/test_transpose.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import print_function, division, absolute_import
+import numpy as np
+import math
+from numba import cuda, float32, float64, complex64, complex128
+from numba.cuda.testing import unittest
+from numba.cuda.cudadrv.driver import driver
+from ddt import ddt, data, file_data, unpack
+
+
+# There ought to be a better way to do this !
+type_map={ np.float32 : float32,
+           np.float64 : float64,
+           np.complex64 : complex64,
+           np.complex128 : complex128}
+
+
+def transpose(a, b):
+    """This implements the algorithm documented in 
```

Comment:
The semantics of this transpose function differ from numpy's transpose function, so it would be helpful for the docstring to explain what the function is doing in terms of its input and output arguments (i.e. stores the transpose of `a` in `b`).